File: Data Structures & Algorithms/three-integer-sum/submission-0.cs
public class Solution {
    public List<List<int>> ThreeSum(int[] nums) {
        Array.Sort(nums);
        var res = new List<List<int>>();

        for (var p0=0;p0<=nums.Length-3;p0++)
        {
            if (p0 > 0 && nums[p0] == nums[p0 - 1])
                continue;
            
            var l=p0+1;
            var r = nums.Length - 1;

            while (l < r)
            {
                if (nums[p0] + nums[l] + nums[r] == 0)
                {
                    res.Add(new List<int> {nums[p0], nums[l], nums[r]});
                    l++;
                    while (nums[l] == nums[l-1] && l < r)
                        l++;
                }
                else if (nums[p0] + nums[l] + nums[r] > 0)
                {
                    r--;
                }
                else
                {
                    l++;
                } 
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:

        leftp, rightp = 1, 1

        ans = nums[0]

        for i in range(len(nums)):
            leftp = (1 if leftp == 0 else leftp) * nums[i]
            rightp = (1 if rightp == 0 else rightp) * nums[len(nums) - i - 1]

            ans = max (ans, leftp, rightp)

        return ans